numpyro/infer/mcmc.py

```diff
@@ -342,12 +346,6 @@ def _single_chain_mcmc(self, rng_key, init_state, init_params, args, kwargs, col
             states[self._sample_field] = lax.map(postprocess_fn, states[self._sample_field])
         return states, last_state
 
-    def _single_chain_jit_args(self, init, collect_fields):
```

Comment:
Hmm..I don't remember why we needed this (or whether we needed it then and there have been some changes in jax since). I am also noticing that we can also remove `sample_fn_jit_args` and `sample_fn_nojit_args` and directly use `partial(sampler_sample, args=, kwargs=)` in `_get_cached_fn(..)` and simplify the state to be `HMCState` instead of a tuple. Let me know if you want me to push this to your branch.

Reply:
I think this method is used to resolved `*init` vs `init` as an argument for `single_chain_mcmc`. I don't remember why we need to distinguish `jit_args` vs `nojit_args` neither. For `cached_fn`, please feel free to push the simplification to this branch. I guess it is needed because if `args`, `kwargs` changed, the cached fn is also needed to be changed.

Reply:
Sorry, I was confused. I don't think we can avoid cycling the `args, kwargs` unfortunately.